golang/thrift/outbound.go

```diff
@@ -0,0 +1,323 @@
+package thrift
+
+import "bytes"
+import "errors"
+import "github.com/apache/thrift/lib/go/thrift"
+import tchannel "github.com/uber/tchannel/golang"
+import "golang.org/x/net/context"
+import "time"
+
+// NewTChannelOutboundProtocol creates a TChannelOutboundProtocol
+func NewTChannelOutboundProtocol(remoteHostPort, remoteServiceName, remoteProcessorName string) (*TChannelOutboundProtocol, error) {
+	tchannel, err := tchannel.NewChannel("0.0.0.0:0", nil)
```

Comment:
@mmihic Thank you for the quick feedback!

The connections are being re-used across calls from the same client. But I'll add a different version of the NewTChannelOutboundProtocol function to take a TChannel object so connections can be re-used across *clients*.